### PR TITLE
Use tags to indicate the pull request status of tasks

### PR DESCRIPTION
**Pull request by ddufresne**
https://isidor.numigi.net/web#active_id=15981&id=15981&model=project.task&view_type=form&menu_id=

### CODE REVIEW

**Comment by ddufresne** [comment]
@foutoucour je crois que tu vas aimer cette proposition. On en avait brièvement discuté lors du numikathon.

**Comment by foutoucour** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- github_pull_request_project/tests/test_project_task.py  1
- github_pull_request_project/models/github_pull_request.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/odoo-git-addons/odoo-git-addons/pullRequest?prid=4509519&bid=15195687)

**Comment by foutoucour** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- github_pull_request_project/tests/test_project_task.py  1
- github_pull_request_project/models/github_pull_request.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/odoo-git-addons/odoo-git-addons/pullRequest?prid=4509519&bid=15195687)

**Comment by foutoucour** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- github_pull_request_project/models/github_pull_request.py  5
- github_pull_request_project/tests/test_project_task.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/odoo-git-addons/odoo-git-addons/pullRequest?prid=4509519&bid=15195687)

File: github_pull_request_project/models/github_pull_request.py [hunk 1]
```diff
@@ -13,18 +16,37 @@ class GithubPullRequestTask(models.Model):
...
+        pr.task_ids._update_pull_request_tags()
+        return pr
 
-def is_no_pull_request_open(task) -> bool:
-    """ Return True if the task has no PR or if all the related PRs are merged or closed."""
-    return (
-        task.pull_request_ids and not task.pull_request_ids.filtered(lambda pr: pr.state == "open")
-        or not task.pull_request_ids
+    @api.multi
+    def write(self, vals):
```
Comment (foutoucour) [inline github_pull_request_project/models/github_pull_request.py:26]: Est ce qu'on pourrait rajouter un `onchange` pour faire que le tag se mette à jour en temps réel ?

Comment (ddufresne) [reply github_pull_request_project/models/github_pull_request.py:26]: bonne idée

Comment (ddufresne) [reply github_pull_request_project/models/github_pull_request.py:26]: @foutoucour done

**Review by foutoucour — APPROVED** [review]

**Comment by foutoucour** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- github_pull_request_project/models/github_pull_request.py  5
- github_pull_request_project/tests/test_project_task.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/odoo-git-addons/odoo-git-addons/pullRequest?prid=4509519&bid=15195687)

**Pull request merged**